pkg/machines/components/storagePools/storagePool.jsx

```diff
@@ -29,68 +29,129 @@ import {
+            </React.Fragment>
+        );
+        const sizeLabel = (
+            <React.Fragment>
+                {`${allocation} / ${capacity} GB`}
+            </React.Fragment>
+        );
+        const state = (
+            <React.Fragment>
+                { this.state.actionError && <span className='pficon-warning-triangle-o machines-status-alert' /> }
+                <span id={`${idPrefix}-state`}>
+                    { storagePool.active ? _("active") : _("inactive") }
+                </span>
+            </React.Fragment>);
+        const cols = [
+            {name, 'header': true},
+            size,
+            sizeLabel,
+            rephraseUI('connections', storagePool.connectionName),
+            state,
+        ];
+
+        const overviewTabName = (
+            <div id={`${idPrefix}-overview`}>
+                {_("Overview")}
+            </div>
+        );
+        const storageVolsTabName = (
+            <div id={`${idPrefix}-storage-volumes`}>
+                {_("Storage Volumes")}
+            </div>
+        );
+        let tabRenderers = [
+            {
+                name: overviewTabName,
+                renderer: StoragePoolOverviewTab,
+                data: {
+                    storagePool, actionError: this.state.actionError,
+                    actionErrorDetail: this.state.actionErrorDetail,
+                    onActionErrorDismiss: () => { this.setState({ actionError:  undefined }) }
+                }
+            },
+            {
+                name: storageVolsTabName,
+                renderer: StoragePoolVolumesTab,
+                data: { storagePool }
+            },
+        ];
+
+        return (
+            <ListingRow rowId={idPrefix}
+                columns={cols}
+                tabRenderers={tabRenderers}
+                listingActions={<StoragePoolActions actionErrorSet={this.actionErrorSet.bind(this)} storagePool={storagePool} />} />
+        );
+    }
+}
 StoragePool.propTypes = {
     storagePool: PropTypes.object.isRequired,
 };
+
+class StoragePoolActions extends React.Component {
+    onUndefine() {
+        storagePoolUndefine(this.props.storagePool.connectionName, this.props.storagePool.id)
+                .fail(exc => {
+                    this.props.actionErrorSet(_("Storage Pool failed to get undefined"), exc.message);
+                });
+    }
+
+    render() {
+        const { storagePool } = this.props;
+        const id = storagePoolId(storagePool.name, storagePool.connectionName);
+        const undefineBtn = (
+            <Button id={`undefine-${id}`} bsStyle='danger' disabled={storagePool.active} onClick={this.onUndefine.bind(this)}>
+                {_("Undefine")}
+            </Button>
+        );
+
+        if (storagePool.active) {
+            return (
+                <OverlayTrigger overlay={ <Tooltip id='tip-pool-undefine'>{_("Try deactivating the storage pool before attempting to undefine it")}</Tooltip> } placement='top'>
```

Comment:
There is no button to deactivate the pool, right? 

**Update**: The following commit adds it.